src/Symfony/Component/HttpFoundation/Response.php

```diff
@@ -1210,6 +1215,19 @@ public static function closeOutputBuffers(int $targetLevel, bool $flush): void
         }
     }
 
+    /**
+     * Mark a response as safe according to RFC8674.
+     *
+     * @see https://tools.ietf.org/html/rfc8674
+     */
+    public function setSafe()
+    {
+        $this->isSafe = true;
+
+        $this->headers->set('Preference-Applied', 'safe');
+        $this->setVary('Prefer', false);
```

Comment:
I'm not sure if I should add support for unsetting safe flag. If so, I'm not sure how to safely remove `Vary` header.

Reply:
I would not remove the "vary" if we define a way to unset the safe flag (unsetting should be possible via an method argument)